src/utils/document-section-utils.ts

```diff
@@ -0,0 +1,268 @@
+      checkedChildSection.push(...getCheckedChildSection(childSection, flatStructure));
+    }
+  }
+
+  return checkedChildSection;
+}
+
+export function toggleAllSectionChildElements(
+  flatStructure: FlatSectionNodeStructure[],
+  sectionId: string,
+  isDisabled: boolean
+): void {
+  const section = flatStructure.find(section => section.id === sectionId);
+
+  if (section?.childrenIds.length && section?.isChecked) {
+    for (const id of section.childrenIds) {
+      toggleDisableInput(id, isDisabled);
+      toggleAllSectionChildElements(flatStructure, id, isDisabled);
+    }
+  }
+}
+
+export function filterDocumentSections(
+  renderedDoc: HTMLElement,
+  nodes: Node[],
+  excludedNodes: string[],
+  storedStyles: StoredStyle[]
+): HTMLElement {
+  const proseMirrorContainer = renderedDoc.getElementsByClassName('ProseMirror')[0] ?? null;
+  const tempTocNodeList = JSON.parse(JSON.stringify(nodes));
+
+  if (proseMirrorContainer) {
+    for (const id of excludedNodes) {
+      const node = nodes.find(node => node.attrs.objectId === id) ?? null;
+
+      if (node?.attrs?.styleName) {
+        const nodeStyle = node.attrs.styleName;
+        const workingIndexes = tempTocNodeList.filter(node => node.attrs.styleName === nodeStyle);
+        const workingSection = workingIndexes.findIndex(node => node.attrs.objectId === id);
+        const sectionElements = proseMirrorContainer.children;
+        const collectionArray = Array.from(sectionElements);
+        const nodeIndexs = getIndexBySectionName(collectionArray, nodeStyle);
+        const startingIndex = nodeIndexs[workingSection];
+        const sectionLevel = getStyleLevel(nodeStyle, storedStyles);
+
+        if (!sectionLevel || !sectionElements[startingIndex]) break;
+
+        sectionElements[startingIndex].remove();
+
+        if (!sectionElements[startingIndex]) break;
+
+        deleteDocumentChildElements(startingIndex, sectionElements, storedStyles);
+      }
+
+      const removedSection = tempTocNodeList.findIndex(node => node.attrs.objectId === id);
+      tempTocNodeList.splice(removedSection, 1);
+    }
+  }
+
+  return renderedDoc;
+}
+
+export function buildListOfIdsToAdd(
+  sectionId: string,
+  currentListOfExcludedIds: string[],
+  flatStructure: FlatSectionNodeStructure[]
+): string [] {
+  const section = flatStructure.find(section => section.id === sectionId);
+  section.isChecked = true;
+
+  const ids = addSelectedSection(flatStructure, sectionId);
+  let newNodeList = JSON.parse(JSON.stringify(currentListOfExcludedIds));
+
+  for (const id of ids) {
+    newNodeList = newNodeList.filter(nodeId => nodeId !== id);
+  }
+
+  return sortExcludeListByFlattenedSection(newNodeList, flatStructure);
+}
+
+export function buildListOfIdsToRemove(
+  sectionId: string,
+  currentListOfExcludedIds: string[],
+  flatStructure: FlatSectionNodeStructure[]
+): string[] {
+  let newNodeList = JSON.parse(JSON.stringify(currentListOfExcludedIds));

```

Comment:
structuredClone? Also, these are strings so why can't you just use spread (shallow clone)?